### PR TITLE
 Add new command `az spring list-support-server-versions` 

### PR DESCRIPTION
**Pull request by yilims**
### Related command
Add new command `az spring list-support-server-versions` to list all supported server versions.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/spring/azext_spring/_transformers.py [hunk 1]
```diff
@@ -365,3 +365,12 @@ def transform_apm_type_output(result):
...
     return result if is_list else result[0]
+
+
+def transform_support_server_versions_output(result):
+    is_list = isinstance(result, list)
+
+    if not is_list:
+        result = [result]
+
+    return result if is_list else result[0]
```
Comment (yuwzho) [inline src/spring/azext_spring/_transformers.py:376]: From my understanding, this method does nothing. Do you wanna extract the important information from the raw data and show as table?

**Review by yuwzho — APPROVED** [review]

File: src/spring/azext_spring/_transformers.py (hunk 1 above)
Comment (yilims) [reply src/spring/azext_spring/_transformers.py:381]: defined table output columns

**Comment by yilims** [comment]
@zhoxing-ms : could you pls help to review & merge the PR?

**Comment by zhoxing-ms** [comment]
Please add some new tests for this PR

**Comment by yilims** [comment]
> Please add some new tests for this PR

@zhoxing-ms : added test case, thanks!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**